packages/components/badge/src/Badge/Badge.tsx

```diff
@@ -32,16 +32,16 @@ export type BadgeInternalProps = CommonProps & {
    */
   endIcon?: React.ReactNode;
   /**
-   * Disables CSS text transforms
+   * Expects any valid CSS text-transform value. If not provided, will default to 'capitalized'
    */
-  noTextTransforms?: boolean;
+  textTransform?: CSSProperties['textTransform'];
 };
 
 export type BadgeProps = PropsWithHTMLElement<BadgeInternalProps, 'div'>;
 
 export const Badge = React.forwardRef<HTMLDivElement, ExpandProps<BadgeProps>>(
   (props, ref) => {
-    const styles = getBadgeStyles(props.noTextTransforms);
+    const styles = getBadgeStyles(props.textTransform ?? 'capitalized');
```

Comment:
Nit: I'd change the parameter in `getBadgeStyles` to be have a default value instead to make sure we always fall back to the same value

```typescript
export const getBadgeStyles = (
  textTransform: CSSProperties['textTransform'] = 'capitalize',
) => ({
```